metricbeat/module/system/process/helper.go

```diff
@@ -373,6 +383,11 @@ func (procStats *ProcStats) GetProcStats() ([]common.MapStr, error) {
 	}
 	procStats.ProcsMap = newProcs
 
+	var summaries *processSummaries
+	if procStats.Summaries {
+		summaries = procStats.getSummaries(processes)
```

Comment:
I'm a little concerned that users might expect the summary to unfiltered. As written it looks like the summary only includes processes that matched the given regular expressions.

If we keep it this way, I feel like we should include info about what we matched against in the summary. This would be needed to distinguish between two separate summary events if the `system/process` metricset is used more than once.